static/app/utils/queryClient.tsx

```diff
@@ -62,6 +62,7 @@ export type ApiQueryKey =
         Record<string, any>,
         Record<string, any>
       >,
+      additionalKey?: string,
```

Comment:
See  [`ca8cc8e` (#86300)](https://github.com/getsentry/sentry/pull/86300/commits/ca8cc8ebd428e9337a95636fcbc13d5a68114b0c#diff-2213a680ea140c136277e40b1150a12d9e9145a0b5aeb78bd0e8d688c2225e53R54-R56)

Reply:
I think the better fix would be to have `useInfiniteApiQuery` _always_ append an additional `’infinite’` key. This would avoid future conflicts if the same endpoint is used as query and infinite query. I think there’s only one other usage of `useInfiniteApiQuery` so it shouldn’t be a huge impact.

Reply:
I agree, though it gets kind of messy because [`queryKey` can be 1 or 2 elements](https://github.com/getsentry/sentry/blob/ref-ui-change-release-series-use-hook-2/static/app/utils/queryClient.tsx#L262)

I suppose we can check the queryKey length and do something like:

```
  const query = useInfiniteQuery({
    queryKey:
      queryKey.length === 1
        ? ([...queryKey, {}, 'infinite'] as const)
        : ([...queryKey, 'infinite'] as const),
```

Reply:
yeah that’s not too bad. What it would then take is to make an `InfiniteApiQueryKey` type that we would use in `fetchInfiniteQuery`. `useInfiniteApiQuery` would still accept an `ApiQueryKey`, and we would do the transformation internally.

I split up the types a bit to get there:

```
type UrlOnly = readonly [url: string];
type UrlWithOptions = readonly [
  url: string,
  options: QueryKeyEndpointOptions<
    Record<string, string>,
    Record<string, any>,
    Record<string, any>
  >,
];

export type ApiQueryKey = UrlOnly | UrlWithOptions;

type InfiniteApiQueryKey = readonly [...UrlWithOptions, {type: 'infinite'}];
```

Reply:
sgtm, will do this in a followup